--- Dockerfile.official	2023-06-01 09:14:26
+++ Dockerfile.multiarch	2023-06-01 09:32:57
@@ -8,7 +8,12 @@
 RUN apt update -y && apt install curl unzip -y
 
 WORKDIR /actions-runner
-RUN curl -f -L -o runner.tar.gz https://github.com/actions/runner/releases/download/v${RUNNER_VERSION}/actions-runner-linux-${RUNNER_ARCH}-${RUNNER_VERSION}.tar.gz \
+
+# Do not use RUNNER_ARCH, because we leave it to avoid conflict on patch
+ARG TARGETARCH
+RUN export RUNNER_ARCH_THIS="x64" \
+    && if [ "$TARGETARCH" = "arm64" ]; then export RUNNER_ARCH_THIS=arm64 ; fi \
+    && curl -f -L -o runner.tar.gz https://github.com/actions/runner/releases/download/v${RUNNER_VERSION}/actions-runner-linux-${RUNNER_ARCH_THIS}-${RUNNER_VERSION}.tar.gz \
     && tar xzf ./runner.tar.gz \
     && rm runner.tar.gz
 
@@ -17,7 +22,7 @@
     && rm runner-container-hooks.zip
 
 RUN export DOCKER_ARCH=x86_64 \
-    && if [ "$RUNNER_ARCH" = "arm64" ]; then export DOCKER_ARCH=aarch64 ; fi \
+    && if [ "$TARGETARCH" = "arm64" ]; then export DOCKER_ARCH=aarch64 ; fi \
     && curl -fLo docker.tgz https://download.docker.com/linux/static/stable/${DOCKER_ARCH}/docker-${DOCKER_VERSION}.tgz \
     && tar zxvf docker.tgz \
     && rm -rf docker.tgz
